duplicating changes from jdpipe/gs-netstream

jdpipe opened the pull request:
I need more info to follow the instructions about pull requests, so I'm making this one anyway!
("with OpenModelica/OMCompiler/3rdParty submodule pointing at your commit" = ?)

https://github.com/graphstream/gs-netstream/pull/10

Note that the OpenModelica workflow **does not use pull requests** for this repository:
https://github.com/OpenModelica/OpenModelica/blob/master/CONTRIBUTING.md

Pull requests are still fine for things like documentation changes that
do not affect building OpenModelica, but there is very little
documentation here.

jdpipe committed:
duplicating changes from jdpipe/gs-netstream

https://github.com/graphstream/gs-netstream/pull/10

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/OpenModelica/OMCompiler-3rdParty?pullRequest=108) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/OpenModelica/OMCompiler-3rdParty?pullRequest=108) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/OpenModelica/OMCompiler-3rdParty?pullRequest=108) it.</sub>

jdpipe committed:
updated gc-8.0.6

jdpipe committed:
Update CMakeLists.txt

gc lib target changed from `omcgc` to `gc-lib` to match `3rdParty/gc/CMakeLists.txt`. let's see...

jdpipe committed:
seems to be needed for building on Ubuntu 20.04 within OpenModelica.

jdpipe committed:
Revert "updated gc-8.0.6"

This reverts commit 7569a3949dc52ea4d859fa4e382529148c93010f.

jdpipe committed:
flexible naming of omcgc vs gc-lib

jdpipe committed:
ignore FMIL/include

jdpipe committed:
Correcting Windows detection for MSYS2.

See https://stackoverflow.com/questions/2989810
and https://stackoverflow.com/questions/17493759

jdpipe committed:
detect windows as _WIN32

jdpipe committed:
add 'long long' in graphstream (needed on MSYS2, apparently)

jdpipe committed:
use int64_t instead of long bc of windows sizes

jdpipe committed:
updating graphstream windows long.

jdpipe committed:
Merge branch 'master' of github.com:jdpipe/OMCompiler-3rdParty into HEAD

jdpipe committed:
Merge branch 'OpenModelica:master' into master
